### ISSUE TITLE
Improve `AsyncWorker` example handling `Object/Buffer/Uint8Array` as parameter

### ISSUE THREAD

**Issue by mahnunchik** [nodejs/node-addon-examples#380]
Hello,

I've created issue in `node-addon-api` https://github.com/nodejs/node-addon-api/issues/1453 but it may be related more to scope of this repository.

>  I couldn't find in the documentation and examples how to properly handle Object/Buffer/Uint8Array parameters in AsyncWorker?

https://github.com/nodejs/node-addon-api/issues/1453

**Comment by legendecas** [comment]
I believe https://github.com/nodejs/node-addon-examples/tree/main/src/4-references-and-handle-scope/function-reference-demo/node-addon-api can be a good example to prevent objects from being reclaimed by the garbage collector.

**Comment by mahnunchik** [comment]
Yes it is good starting point, but working with `AsyncWorker` is a bit different because it was not clear to me where to wrap and where to release data. Now it is clear for me https://github.com/nodejs/node-addon-api/issues/1453#issuecomment-1970549421 but I think example can be added.
